### PR TITLE
Adding a hiring process page

### PR DESCRIPTION
**Pull request by eiso**
The first commit is mainly scaffolding, @tsolakoua will be filling this out based on our meeting (https://github.com/src-d/minutes/pull/93).

### CODE REVIEW

**Comment by campoy** [comment]
This fixes #83

**Comment by eiso** [comment]
Good idea @campoy this was originally to outline our internal process only but we can add some sections for external as well. I'm waiting for @tsolakoua's commits and then we can review and add one. This is still a bare PR.

**Comment by tsolakoua** [comment]
@eiso just added the content from my side. 
@campoy agree with Eiso. The content is focused on the internal process of hiring and the responsibilities are being given.

File: talent/hiring_process.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Hiring Process Explained
+
+This document explains the internal hiring process at source{d} and the responsibilities for each person related to it.
+
+
+### Job Description
+
+- Job Description is written by the lead engineers with Anna's support.
```
Comment (dpordomingo) [inline talent/hiring_process.md]: Is the text initially written publicly, gdocs, any src-d repo?

File: talent/hiring_process.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+- The candidate has no time deadline to accept/reject the offer. Only in cases that we have more candidates in the offer state for one opening where decisions need to be made. 
+
+## Notes 
+
+### Exceptions 
+
+- There are exceptions in the above steps for internship roles. The first step in this case is the Coding Challenges and then a technical Skype interview. 
+ 
+ ### Feedback
```
Comment (dpordomingo) [inline talent/hiring_process.md]: I think there is a typo: leading space

File: talent/hiring_process.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+- Every candidate who is passing the first Skype interview, should complete the coding challege. 
+
+- In case of Data Science, there is an additional ML task necessary to be solved.
+
+- Coding Challeges are being sent to the candidates by Anna and the results are being reviewed by the lead engineers. 
+
+### On-site Interview
+
+- This is the last step in which it's being decided if the candidate will receive an offer or not. It might last a full day or being split in two days, regarding the availability of the candidate. 
+
```
Comment (dpordomingo) [inline talent/hiring_process.md]: Since this is going to be public. Do you think it would be a good idea to mention the three different interviews that will take place? Or is this something to be avoided?
For example
- **Q&A** to know each other,
- **Design interview** to see how the candidate would design a solution to a given technical problem working with a couple of our engineers
-  **Logical thinking** to see how the candidate solves an abstract problem in a logical way.

**Comment by eiso** [comment]
@tsolakoua it's looking good! I made some minor changes. Before we PR, I'd like to replace your name everywhere with the job title, and a bio about you on-top of the document. Also, I would like to replace all mentions of candidate to developer if you agree. These last two suggestions, I haven't made as changes and will leave with you.

File: talent/hiring_process.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+- The Talent Manager is responsible to work with the Developer Community team in order to gain people through community actions (Twitter/Blog etc.). 
+
+### First Contact
+
+- Any developer should be contacted by the lead engineers or under their name at least (after together defining the outgoing message). In case the Talent Manager is contacting them, it should be done from the lead's email acccount. It's a motivating factor for the developers to be contacted by the lead engineers. 
+
+### Screening
+
+- If the developer is an applicant, they should be accepted/rejected by the lead engineers during the screening phase.
```
Comment (ajnavarro) [inline talent/hiring_process.md]: Any applicant should have an interview with the lead? maybe we can limit this depending of the source of the applicant. If we know that the source is worth it, then for me is ok to do the first interview, but if the applicant can be anyone, then I would prefer to apply a filter first.

File: talent/hiring_process.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+- At any given moment the Talent Manager is able to drop in and work on finding more developers.
+
+- Developer lists/sourcing channels should continuously be defined and refreshed after communication between the Talent Manager and the leads. 
+
+- The Talent Manager is responsible to work with the Developer Community team in order to gain people through community actions (Twitter/Blog etc.). 
+
+### First Contact
+
+- Any developer should be contacted by the lead engineers or under their name at least (after together defining the outgoing message). In case the Talent Manager is contacting them, it should be done from the lead's email acccount. It's a motivating factor for the developers to be contacted by the lead engineers.
```
Comment (rporres) [inline talent/hiring_process.md]: I'm not entirely comfortable having mails sent using my mail account. I find it weird...

File: talent/hiring_process.md (hunk 4 above)
Comment (rporres) [reply talent/hiring_process.md]: I think it is a good idea to have the lead doing a first screening, to present the company, the job, talk with the candidate about their past and find any potential red flag.

File: talent/hiring_process.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+- Coding Challeges are being sent to the candidates by Anna and the results are being reviewed by the lead engineers. 
+
+### On-site Interview
+ 
+- This is the last step in which it's  decided if the candidate will receive an offer or not. It might last a full day or be split across two half-days, depending on the availability of the candidate. 
+
+- The interviews are the followings: 
+
+* Kickoff with Maximo, QA: to know the candidate better
+* Design interview: to see how the candidate would design a solution to a given technical problem working with a several of our engineers
```
Comment (rporres) [inline talent/hiring_process.md]: In case of a candidate being interviewed for infrastructure we have to make sure that design interview covers what we need. It has happened in the past that I had to make an additional infra interview as design was too wide (candidate was also considered for applications team)

File: talent/hiring_process.md (hunk 4 above)
Comment (bzz) [reply talent/hiring_process.md]: I'm not sure how debatable this point is, but personally I agree \w @ajnavarro - it could be lead by default, but in some cases should be possible to be handled by Talent Manager as well.

File: talent/hiring_process.md (hunk 5 above)
Comment (bzz) [reply talent/hiring_process.md]: Not a strong opinion, but personally that is a bit strange for me as well. At this point, applicant would have known nothing about a lead or a team, so I'm not sure how motivating can be to know that somebody, you will be doing technical work with, is spending time sending emails.

At the same time I'm enthusiastic 👍 for leads proactively reaching out to potential candidates they know about.

File: talent/hiring_process.md (hunk 4 above)
Comment (eiso) [reply talent/hiring_process.md]: @tsolakoua will ask the leads to review every application in Lever (so no calls needed here), that isn't an obvious no (still quite often we get applications where there is an obvious mismatch between the requirements and the person). If this person's application looks promising we expect the leads to do the first call for exactly the reasons @rporres mentions. 

Know that for the vast majority of first calls you're doing they are coming from the original list of people you've identified, so in the end, this shouldn't be a big burden. If it becomes one, we can revisit this topic.

File: talent/hiring_process.md (hunk 5 above)
Comment (eiso) [reply talent/hiring_process.md]: @rporres very simple, you send them out yourself. This is purely a support offer from @tsolakoua in case you don't have the time and want to reach out to a larger number of people with similar messaging.

File: talent/hiring_process.md (hunk 6 above)
Comment (eiso) [reply talent/hiring_process.md]: This is a very good point! Let's add a reference to this to the guide @tsolakoua

File: talent/hiring_process.md (hunk 5 above)
Comment (rporres) [reply talent/hiring_process.md]: If you have a strong opinion about mails coming from the lead, I can live with that. But I agree with @bzz about the doubts of how motivating can be to receive mails from the lead. I do agree that it is that a phone screen with the lead is certainly motivating but I don't feel the same with mails.

File: talent/hiring_process.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+- If the developer is an applicant, they should be accepted/rejected by the lead engineers during the screening phase. 
+
+- If the developer is approached by us from a specific source (known from meetups, having worked together in the past,  etc.), they should move forward to the next step as it means that the leads have already shown interest.
+
+### First Skype/Zoom/Phone Interview
+
+- A quick Skype/Phone/Zoom call should be available to every developer who is interested in the roles*. It is a general discussion to know more about the background and experience of the developer. And to share as much as possible from our side about the team & company. 
+
+\* There a noteable exceptions to this, such as higher volume junior roles where we directly ask the developer to do the coding challenge before a call.
```
Comment (vmarkovtsev) [inline talent/hiring_process.md]: There **are**?

File: talent/hiring_process.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+- Lead engineers are responsible to make the calls as it's a motivating factor for the developers. Leads will also be able to assess early in the process if the interview process should continue or not. 
+
+### Coding Challenges 
+
+- Every developer who passes the first Skype interview, should complete the coding challege. 
+
+- In case of the Machine Learning team, there is an additional ML task necessary to be solved.
+
+- Coding Challeges are being sent to the developers by the Talent Manager and the results are being reviewed by the lead engineers.
```
Comment (vmarkovtsev) [inline talent/hiring_process.md]: Challenges

File: talent/hiring_process.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+- Lead engineers are responsible to make the calls as it's a motivating factor for the developers. Leads will also be able to assess early in the process if the interview process should continue or not. 
+
+### Coding Challenges 
+
+- Every developer who passes the first Skype interview, should complete the coding challege. 
+
+- In case of the Machine Learning team, there is an additional ML task necessary to be solved.
+
+- Coding Challeges are being sent to the developers by the Talent Manager and the results are being reviewed by the lead engineers. 
+
```
Comment (vmarkovtsev) [inline talent/hiring_process.md]: We had some "second chance" exceptions in the past. When the guy was super disappointed and showed his motivation and cried out loud, we sent the second coding challenge.

File: talent/hiring_process.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+### On-site Interview
+ 
+- This is the last step in which it's  decided if the developer will receive an offer or not. It might last a full day or be split across two half-days, depending on the availability of the developer. 
+
+- The interviews are the followings:
```
Comment (vmarkovtsev) [inline talent/hiring_process.md]: following

File: talent/hiring_process.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+- Coding Challeges are being sent to the developers by the Talent Manager and the results are being reviewed by the lead engineers. 
+
+### On-site Interview
+ 
+- This is the last step in which it's  decided if the developer will receive an offer or not. It might last a full day or be split across two half-days, depending on the availability of the developer. 
+
+- The interviews are the followings: 
+
+* Kickoff with Maximo, QA: to know the developer better
```
Comment (vmarkovtsev) [inline talent/hiring_process.md]: Máximo

File: talent/hiring_process.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+* Design interview: to see how the developer would design a solution to a given technical problem working with a several of our engineers. In case of a developer being interviewed for the Infrastructure team, we have to make sure that this interview covers the specified needs for the team.
+* Logical thinking: to see how the developer solves an abstract problem in a logical way.
```
Comment (vmarkovtsev) [inline talent/hiring_process.md]: ...and behave in situations where they are exhausted of answers.

File: talent/hiring_process.md [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+
+- The developer has no time deadline to accept/reject the offer. Only in cases that we have more developers in the offer state for one opening where decisions need to be made. 
+
+## Notes 
+
+### Exceptions 
+
+- There are exceptions to the above steps for internship roles. The first step in this case is the Coding Challenges and then a technical Skype interview. 
+ 
+ ### Feedback
```
Comment (vmarkovtsev) [inline talent/hiring_process.md]: Whitespace on the left

File: talent/hiring_process.md (hunk 8 above)
Comment (tsolakoua) [reply talent/hiring_process.md]: I have it as Coding Challenges as there are more than one in case of ML

File: talent/hiring_process.md (hunk 9 above)
Comment (tsolakoua) [reply talent/hiring_process.md]: As this will be posted in the guide which is public, if the candidates read it they might all request a second chance. Also even if we do give second chance (for very specific reasons), it doesn't seem a consistent approach and not sure if it's a good idea to be in this guide.

Comment (vmarkovtsev) [reply talent/hiring_process.md]: Right. Stupid transparency

File: talent/hiring_process.md (hunk 8 above)
Comment (vmarkovtsev) [reply talent/hiring_process.md]: I meant the misprint :)

File: talent/hiring_process.md (hunk 13 above)
Comment (vmarkovtsev) [reply talent/hiring_process.md]: Need to remove the whitespace on the left

File: talent/hiring_process.md [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+- The offer is being sent by the CTO on Lever. 
+
+- The developer has no time deadline to accept/reject the offer. Only in cases that we have more developers in the offer state for one opening where decisions need to be made. 
+
+## Notes 
+
+### Exceptions 
+
+- There are exceptions to the above steps for internship roles. The first step in this case is the Coding Challenges and then a technical Skype interview.
```
Comment (vmarkovtsev) [inline talent/hiring_process.md]: We normally did not do on-sites for interns unless we thought one could be a full-time engineer. Need to ensure this with the management team though.

Comment (eiso) [reply talent/hiring_process.md]: I can confirm this.

File: talent/hiring_process.md [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+This document explains the internal hiring process at source{d} and the responsibilities for each person related to it.
+
+### Job Description
+
+- Job Description is written by the lead engineers with the Talent's Manager support. 
+
+- For every meeting between the lead and the Talent Manager there will be a PR in the [minutes](https://github.com/src-d/minutes) repo and the job descriptions will be uploaded at the private repository https://github.com/src-d/company.
+
+- CTO and CEO should be added as reviewers (but should not be bottlenecks to continue).
```
Comment (campoy) [inline talent/hiring_process.md]: bottlenecks? that implies it's a capacity problem
what does this mean exactly?

File: talent/hiring_process.md [hunk 16]
```diff
@@ -0,0 +1,77 @@
+# Hiring Process Explained
+
+This document explains the internal hiring process at source{d} and the responsibilities for each person related to it.
+
+### Job Description
+
+- Job Description is written by the lead engineers with the Talent's Manager support.
```
Comment (campoy) [inline talent/hiring_process.md]: I'd use "The `Job Description` is written" or "Job Descriptions are written by"

File: talent/hiring_process.md [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+
+- For every meeting between the lead and the Talent Manager there will be a PR in the [minutes](https://github.com/src-d/minutes) repo and the job descriptions will be uploaded at the private repository https://github.com/src-d/company.
+
+- CTO and CEO should be added as reviewers (but should not be bottlenecks to continue).
+
+- The Talent Manager posts the job description through Lever on the source{d} website.
+
+### Sourcing
+
+- Lead engineers are responsible to identity people to be contacted and other sourcing channels such as niche job boards.
```
Comment (campoy) [inline talent/hiring_process.md]: responsible for identifying

File: talent/hiring_process.md [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+### Sourcing
+
+- Lead engineers are responsible to identity people to be contacted and other sourcing channels such as niche job boards. 
+
+- At any given moment the Talent Manager is able to drop in and work on finding more developers.
+
+- Developer lists/sourcing channels should continuously be defined and refreshed after communication between the Talent Manager and the leads. 
+
+- The Talent Manager is responsible to work with the Developer Community team in order to gain people through community actions (Twitter/Blog etc.).
```
Comment (campoy) [inline talent/hiring_process.md]: responsible for working

Comment (campoy) [inline talent/hiring_process.md]: Is Developer Community the same as Developer Relations?

File: talent/hiring_process.md (hunk 5 above)
Comment (campoy) [reply talent/hiring_process.md]: I'd rather have emails sent actually by the leads, or if they're not able to do so by someone else in their names (but not using their names)

File: talent/hiring_process.md [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+
+- If the developer is an applicant, they should be accepted/rejected by the lead engineers during the screening phase. 
+
+- If the developer is approached by us from a specific source (known from meetups, having worked together in the past,  etc.), they should move forward to the next step as it means that the leads have already shown interest.
+
+### First Skype/Zoom/Phone Interview
+
+- A quick Skype/Phone/Zoom call should be available to every developer who is interested in the roles*. It is a general discussion to know more about the background and experience of the developer. And to share as much as possible from our side about the team & company. 
+
+\* There are noteable exceptions to this, such as higher volume junior roles where we directly ask the developer to do the coding challenge before a call.
```
Comment (campoy) [inline talent/hiring_process.md]: notable

File: talent/hiring_process.md [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+
+### Screening
+
+- If the developer is an applicant, they should be accepted/rejected by the lead engineers during the screening phase. 
+
+- If the developer is approached by us from a specific source (known from meetups, having worked together in the past,  etc.), they should move forward to the next step as it means that the leads have already shown interest.
+
+### First Skype/Zoom/Phone Interview
+
+- A quick Skype/Phone/Zoom call should be available to every developer who is interested in the roles*. It is a general discussion to know more about the background and experience of the developer. And to share as much as possible from our side about the team & company.
```
Comment (campoy) [inline talent/hiring_process.md]: The last sentence should be merged to the previous one, or simply rephrased into something else.

File: talent/hiring_process.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+
+- Coding Challenges are being sent to the developers by the Talent Manager and the results are being reviewed by the lead engineers. 
+
+### On-site Interview
+ 
+- This is the last step in which it's  decided if the developer will receive an offer or not. It might last a full day or be split across two half-days, depending on the availability of the developer. 
+
+- The interviews are the following: 
+
+* Kickoff with the CTO, QA: to know the developer better
```
Comment (campoy) [inline talent/hiring_process.md]: End with a `.`

File: talent/hiring_process.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+* Design interview: to see how the developer would design a solution to a given technical problem working with a several of our engineers. In case of a developer being interviewed for the Infrastructure team, we have to make sure that this interview covers the specified needs for the team.
```
Comment (campoy) [inline talent/hiring_process.md]: "with several of our engineers"

Comment (campoy) [reply talent/hiring_process.md]: Why only the Infrastructure?
I'd say each time has specific needs that should be assured before accepting a candidate.

File: talent/hiring_process.md [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+* Logical thinking: to see how the developer solves an abstract problem in a logical way.
+
+- In case of rejection, the developer should be notified in person on the day by the lead of the corresponding team on why he was rejected. This should be a constructive discussion giving the developer as much insights as possible.
```
Comment (campoy) [inline talent/hiring_process.md]: Just a note that this is something quite unusual for Silicon Valley.

File: talent/hiring_process.md [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+
+### Offer
+
+- The offer is being sent by the CTO on Lever. 
+
+- The developer has no time deadline to accept/reject the offer. Only in cases that we have more developers in the offer state for one opening where decisions need to be made.
```
Comment (campoy) [inline talent/hiring_process.md]: The second sentence is very hard to parse. Maybe:

> An exception to this rule is when multiple developers are being considered for an offer.

File: talent/hiring_process.md [hunk 25]
```diff
@@ -0,0 +1,77 @@
...
+
+- The offer is being sent by the CTO on Lever. 
+
+- The developer has no time deadline to accept/reject the offer. Only in cases that we have more developers in the offer state for one opening where decisions need to be made. 
+
+## Notes 
+
+### Exceptions 
+
+- There are exceptions to the above steps for internship roles. The first step in this case is the Coding Challenges and then a technical Skype interview. Also, we do not do on-sites interviews for interns unless one could join us as a full-time engineer.
```
Comment (campoy) [inline talent/hiring_process.md]: on-site interviews

File: talent/hiring_process.md [hunk 26]
```diff
@@ -0,0 +1,77 @@
...
+ 
+### Feedback
+
+ - After the coding challenge it's very important to gain feedback of the CTO at each next step.
```
Comment (campoy) [inline talent/hiring_process.md]: s/gain feedback of/obtain feedback from/

File: talent/hiring_process.md (hunk 15 above)
Comment (tsolakoua) [reply talent/hiring_process.md]: Means to delay and pause the process. Will rephrase it.

File: talent/hiring_process.md (hunk 18 above)
Comment (tsolakoua) [reply talent/hiring_process.md]: Yes. Changed

File: talent/hiring_process.md (hunk 22 above)
Comment (tsolakoua) [reply talent/hiring_process.md]: I think Applications, ML and Data Retrieval have similar problems to give to the candidate as there is intersection between them, comparing to the Infrastructure. But writing in a more general way, making sure that the needs of should be assured for every team covers the issue that had been faced by Infrastructure team.

File: talent/hiring_process.md (hunk 23 above)
Comment (eiso) [reply talent/hiring_process.md]: My understanding that in the US this isn't done for legal liability concerns.
